Mouse and Keyboard related mappings

Ourten opened the pull request:
Map Mouse, Keyboard classes and a few related utils.

Ourten committed:
Map Mouse and Keyboard

asiekierka commented on mappings/net/minecraft/client/util/GLFWUtil.mapping:
```diff
@@ -0,0 +1,2 @@
+CLASS cnj net/minecraft/client/util/GLFWUtil
+	METHOD b getGLFWTime ()D
```
Why not just getTime? It's in GLFWUtil.

Gegy replied:
Should this also specify the unit of time?

Ourten replied:
I've renamed it to getTime but specifying the unit of time should be in a javadoc since it is platform dependent and specific to the GLFW timer.

asiekierka commented on mappings/net/minecraft/client/util/Keyboard.mapping:
```diff
@@ -2,6 +2,10 @@ CLASS cpj net/minecraft/client/util/Keyboard
+	FIELD d debugCrashStartTime J
+	FIELD e debugCrashLastLogTime J
+	FIELD f debugCrashCounter J
+	FIELD g switchF3State Z
```
Is this hardcoded to F3?

shadowfacts replied:
Not F3, but it's hardcoded to F3+C and only used for debug crashing. Perhaps `switchDebugCrashState`?

Ourten replied:
It's also used to prevent the toggle of the F3 screen when a F3+[something] shortcut is made.

asiekierka commented on mappings/net/minecraft/client/util/Mouse.mapping:
```diff
@@ -1,7 +1,19 @@
+	FIELD m cursorYSmoother Lyz;
+	FIELD n cursorDeltaX D
+	FIELD o cursorDeltaY D
+	FIELD p eventDWheel D
```
eventDeltaWheel?

shadowfacts commented on mappings/net/minecraft/client/MinecraftClient.mapping:
```diff
@@ -116,6 +116,7 @@ CLASS cpk net/minecraft/client/MinecraftClient
 	METHOD ag hasReducedDebugInfo ()Z
 	METHOD ah getToastManager ()Lcrp;
 	METHOD ai getTutorialManager ()Ldrd;
+	METHOD aj isWindowFocused ()Z
```
The setter and the backing field for this should also be mapped.

shadowfacts commented on mappings/net/minecraft/client/util/Keyboard.mapping:
```diff
@@ -2,6 +2,10 @@ CLASS cpj net/minecraft/client/util/Keyboard
 	FIELD c clipboard Lcns;
+	FIELD d debugCrashStartTime J
+	FIELD e debugCrashLastLogTime J
+	FIELD f debugCrashCounter J
```
What specifically does this count? It's not immediately clear from the code.

Ourten replied:
It keep track of the seconds elapsed since the beginning of the countdown but is not really used for anything relevant.

shadowfacts replied:
Is there a name that better reflects this? `debugCrashElapsedTime`? Just counter is fine, but not great.

Ourten replied:
Yes, `debugCrashElapsedTime` is better. I changed it.

Ourten committed:
Simplify mappings and add missing windowFocused mapping

Change getGLFWTime to getTime, Map getter and setter of windowFocused and rename eventDWheel

Ourten committed:
Change debugCrashCounter to debugCrashElapsedTime
